src/test/java/uk/gov/companieshouse/controller/DissolutionDirectorControllerTest.java

```diff
@@ -0,0 +1,172 @@
+
+    private static final String AUTHORISED_USER_HEADER = "ERIC-Authorised-User";
+
+    private static final String COMPANY_NUMBER = "12345678";
+    private static final String USER_ID = "1234";
+    private static final String OFFICER_ID = "abc123";
+    private static final String EMAIL = "user@mail.com";
+
+    @MockBean
+    private DissolutionDirectorService service;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void patchDissolutionDirectorRequest_returnsUnprocessableEntity_ifNoEmailProvided() throws Exception {
+        final DissolutionDirectorPatchRequest body = generateDissolutionPatchDirectorRequest();
+        body.setEmail(null);
+
+        assertPatchDirectorBodyValidation(body, "{'email':'must not be blank'}");
+    }
+
+    @Test
+    public void patchDissolutionDirectorRequest_returnsUnprocessableEntity_ifEmailIsWrongFormat() throws Exception {
+        final DissolutionDirectorPatchRequest body = generateDissolutionPatchDirectorRequest();
+        body.setEmail("wrongemail");
+
+        assertPatchDirectorBodyValidation(body, "{'email':'must be a well-formed email address'}");
+    }
+
+    @Test
+    public void patchDissolutionDirectorRequest_returnsNotFound_ifDirectorOrDissolutionDoesntExist() throws Exception {
+        final DissolutionDirectorPatchRequest body = generateDissolutionPatchDirectorRequest();
+
+        when(service.doesDirectorExist(COMPANY_NUMBER, OFFICER_ID)).thenReturn(false);
+
+        mockMvc
+                .perform(
+                        patch(DIRECTOR_URI, COMPANY_NUMBER, OFFICER_ID)
+                                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                .headers(createHttpHeaders())
+                                .content(asJsonString(body))
+                )
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void patchDissolutionDirectorRequest_returnsBadRequest_ifDirectorNotPendingApproval() throws Exception {
+        final DissolutionDirectorPatchRequest body = generateDissolutionPatchDirectorRequest();
+
+        when(service.doesDirectorExist(COMPANY_NUMBER, OFFICER_ID)).thenReturn(true);
+        when(service.checkPatchDirectorConstraints(COMPANY_NUMBER, OFFICER_ID, EMAIL)).thenReturn(Optional.of("Director is not pending approval"));
+
+        mockMvc
+                .perform(
+                        patch(DIRECTOR_URI, COMPANY_NUMBER, OFFICER_ID)
+                                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                                .headers(createHttpHeaders())
+                                .content(asJsonString(body)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void patchDissolutionDirectorRequest_returnsBadRequest_ifRequesterIsNotApplicant() throws Exception {
```

Comment:
If requester is not applicant

This isn't handled in the controller and shouldn't be tested in the controller. Instead, it should just say "when a patch constraint fails".